doc/manuals/contribution_guide.md

```diff
@@ -0,0 +1,674 @@
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: <the author>
+*/
+```
+
+For Python, bash script, CMakeLists.txt, etc.:
+
+```
+# Copyright 20XX Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+#
+# Author: <the author>
+```
+
+*Rationale*: to have an homogenous copyright header for all files.
+
+*How to check*": manually?, make lint?, other? TBD
+
+#### M3 (Function header)
+
+*Rule*: All functions MUST have a header, which SHOULD have a short description of what the function does, a 
+descriptive list of its parameters, and its return values.
+
+Example:
+
+```    
+/* ****************************************************************************
+*
+* parseUrl - parse a URL and return its pieces
+*
+*  [ Short description if necessary ]
+*
+* PARAMETERS
+*   - url         The URL to be examined
+*   - host        To output the HOST of the URL 
+*   - port        To output the PORT of the URL
+*   - path        To output the PATH of the URL
+*   - protocol    To output the PROTOCOL of the URL
+*
+* RETURN VALUE
+*   parseUrl returns TRUE on successful operation, FALSE otherwise
+*
+* NOTE
+*   About the components in a URL: according to 
+*   https://tools.ietf.org/html/rfc3986#section-3,
+*   the scheme component is mandatory, i.e. the 'http://' or 'https://' must 
+*   be present, otherwise the URL is invalid.
+*/
+```
+
+*Rationale*: the code is simply easier to read when prepared like this
+
+*How to check*": manually?, make lint?, other? TBD
```

Comment:
After @kzangeli 's last commit, all the rules has the "How to check" correctly filled.

Improving automatic procedures to become all the remainer "manually" to somekind of "checked automatically doing X" is left as future work  (maybe a good item for the bounty program?).